bin/main.rb

```diff
@@ -1,2 +1,71 @@
-#!/usr/bin/env ruby
-puts 'Hello world!'
+# !/usr/bin/env ruby
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/MethodLength
+
+require_relative '../lib/board.rb'
+require_relative '../lib/players.rb'
+require_relative '../lib/logic.rb'
+
+@board = Board.new
+@winner = ''
+@turn = 'X'
+
+def take_turn(player, cell)
+  @board.board[cell] = player
+end
+
+def play
+  @board.display_board
+  puts 'Player1 write your name'
+  temp_player_name = gets.chomp
+  player1 = Player.new(temp_player_name)
+  puts "player1's name is #{player1.name}"
+  puts ''
+  puts 'Player2 write your name'
+  temp_player_name = gets.chomp
+  player2 = Player.new(temp_player_name)
+  puts "Player2's name is #{player2.name}"
+
+  while @winner == ''
+    @board.display_board
+    if @turn == 'X'
+      puts "#{player1.name} select your cell"
+    else
+      puts "#{player2.name} select your cell"
+    end
+    cell = gets.chomp
+
+    while @board.board[cell.to_i] != ' '
+      puts 'Please only select an empty cell between 0 and 8'
+      puts 'Select another cell'
+      cell = gets.chomp
+    end
+
+    take_turn(@turn, cell.to_i)
+
+    @winner = @turn if check_win(@board.board) == true
+
+    @winner = 'T' if @winner == '' && !@board.board.include?(' ')
+
+    @turn = if @turn == 'X'
+              'O'
+            else
+              'X'
+            end
```

Comment:
Improve your writing code.
It's similar to switching the player, move into `Board` class